api/pkg/backend/service.go

```diff
@@ -469,6 +473,63 @@ func (s *APIService) CheckInvalidBackends(ctx context.Context, backends []string
 	return invalidBackends, nil
 }
 
+//response of projects
+type ProjectsResponse struct {
+	Links struct {
+		Self     string      `json:"self"`
+		Previous interface{} `json:"previous"`
+		Next     interface{} `json:"next"`
+	} `json:"links"`
+	Projects []struct {
+		IsDomain    bool   `json:"is_domain"`
+		Description string `json:"description"`
+		Links       struct {
+			Self string `json:"self"`
+		} `json:"links"`
+		Tags     []interface{} `json:"tags"`
+		Enabled  bool          `json:"enabled"`
+		ID       string        `json:"id"`
+		ParentID string        `json:"parent_id"`
+		DomainID string        `json:"domain_id"`
+		Name     string        `json:"name"`
+	} `json:"projects"`
+}
+
+///function to validate tenant:
+func ValidateTenant(authToken, tenantId string) bool {
+	osAuthUrl := os.Getenv("OS_AUTH_URL")
```

Comment:
Is this reallly required to get OS_AUTH_URL through environment variable?

Reply:
yes to get host_ip